papers/views.py

```diff
@@ -124,6 +124,28 @@ def get(self, request, *args, **kwargs):
             request.GET.pop('visible', None)
             request.GET.pop('availability', None)
             request.GET.pop('oa_status', None)
+
+        # Handle search by ORCID
+        query = self.get_form_kwargs()['data']
+        if 'authors' in query:
+            orcids = [
+                item
+                for item in query['authors'].split()
+                if validate_orcid(item)
+            ]
+            for orcid in orcids:
+                try:
+                    researcher = Researcher.objects.get(orcid=orcid)
+                    researchers.append(researcher.id)
+                except MetadaSourceException:
```

Comment:
I would expect this exception to be `Researcher.DoesNotExist` instead.

Reply:
Oh, OK. This is copy pasted from https://github.com/dissemin/dissemin/blob/master/papers/views.py#L212-L219 actually.

Reply:
This exception is thrown by `researcher.init_from_orcid()` which you have removed. Maybe it would make sense to refactor the code you copied into a function that can be reused for both views?

Reply:
I am not sure though - the views could be doing different things. The question is: do we want to trigger fetches from ORCID in Celery if the ORCID id supplied is not in dissemin yet? We do that for the https://dissem.in/<orcid> view, but not sure if we want to put that in the search interface itself…

Reply:
Unrelated but there's a typo: "MetadaSourceException" should be "MetadataSourceException". Note that this typo is not in https://github.com/dissemin/dissemin/blob/master/papers/views.py#L212-L219

Reply:
Fixed and tested.